diff --git a/src/ngx_http_lua_time.c b/src/ngx_http_lua_time.c
index 3272a75..d6ff7e4 100644
--- a/src/ngx_http_lua_time.c
+++ b/src/ngx_http_lua_time.c
@@ -17,6 +17,7 @@

 static int ngx_http_lua_ngx_today(lua_State *L);
 static int ngx_http_lua_ngx_time(lua_State *L);
+static int ngx_http_lua_ngx_usec_time(lua_State *L);
 static int ngx_http_lua_ngx_now(lua_State *L);
 static int ngx_http_lua_ngx_localtime(lua_State *L);
 static int ngx_http_lua_ngx_utctime(lua_State *L);
@@ -75,6 +76,19 @@ ngx_http_lua_ngx_time(lua_State *L)


 static int
+ngx_http_lua_ngx_usec_time(lua_State *L)
+{
+    ngx_time_t              *tp;
+
+    tp = ngx_timeofday();
+
+    lua_pushnumber(L, (lua_Number) tp->usec);
+
+    return 1;
+}
+
+
+static int
 ngx_http_lua_ngx_now(lua_State *L)
 {
     ngx_time_t              *tp;
@@ -217,6 +231,9 @@ ngx_http_lua_inject_time_api(lua_State *L)
     lua_pushcfunction(L, ngx_http_lua_ngx_time);
     lua_setfield(L, -2, "time");

+    lua_pushcfunction(L, ngx_http_lua_ngx_usec_time);
+    lua_setfield(L, -2, "usec_time");
+
     lua_pushcfunction(L, ngx_http_lua_ngx_now);
     lua_setfield(L, -2, "now");
